src/System.Text.Json/src/System/Text/Json/Node/JsonNumber.cs

```diff
@@ -0,0 +1,640 @@
+        public static implicit operator JsonNumber(float value) => new JsonNumber(value);
+
+        /// <summary>
+        ///    Converts a <see cref="double"/> to a JSON number.
+        /// </summary>
+        /// <param name="value">The value to convert.</param>
+        public static implicit operator JsonNumber(double value) => new JsonNumber(value);
+
+        /// <summary>
+        ///    Converts a <see cref="sbyte"/> to a JSON number.
+        /// </summary>
+        /// <param name="value">The value to convert.</param>
+        [CLSCompliant(false)]
+        public static implicit operator JsonNumber(sbyte value) => new JsonNumber(value);
+
+        /// <summary>
+        ///    Converts a <see cref="ushort"/> to a JSON number.
+        /// </summary>
+        /// <param name="value">The value to convert.</param>
+        [CLSCompliant(false)]
+        public static implicit operator JsonNumber(ushort value) => new JsonNumber(value);
+
+        /// <summary>
+        ///    Converts a <see cref="uint"/> to a JSON number.
+        /// </summary>
+        /// <param name="value">The value to convert.</param>
+        [CLSCompliant(false)]
+        public static implicit operator JsonNumber(uint value) => new JsonNumber(value);
+
+        /// <summary>
+        ///    Converts a <see cref="ulong"/> to a JSON number.
+        /// </summary>
+        /// <param name="value">The value to convert.</param>
+        [CLSCompliant(false)]
+        public static implicit operator JsonNumber(ulong value) => new JsonNumber(value);
+
+        /// <summary>
+        ///    Converts a <see cref="decimal"/> to a JSON number.
+        /// </summary>
+        /// <param name="value">The value to convert.</param>
+        public static implicit operator JsonNumber(decimal value) => new JsonNumber(value);
+
+        /// <summary>
+        ///   Compares <paramref name="obj"/> to the value of this instance. 
+        /// </summary>
+        /// <param name="obj">The object to compare against.</param>
+        /// <returns>
+        ///   <see langword="true"/> if the value of this instance matches exactly <paramref name="obj"/> (is equal and has the same format),
+        ///   <see langword="false"/> otherwise.
+        /// </returns>
+        public override bool Equals(object obj) => obj is JsonNumber number && _value == number._value;
+
+        /// <summary>
+        ///   Calculates a hash code of this instance.
+        /// </summary>
+        /// <returns>A hash code for this instance.</returns>
+        public override int GetHashCode() => _value.GetHashCode();
+
+        /// <summary>
+        ///   Compares other JSON number to the value of this instance. 
+        /// </summary>
+        /// <param name="other">The JSON number to compare against.</param>
+        /// <returns>
+        ///   <see langword="true"/> if the value of this instance matches exactly <paramref name="other"/> (is equal and has the same format),
```

Comment:
Same nit here:
nit:
```suggestion
        ///   <see langword="true"/> if the value of this instance matches <paramref name="other"/> exactly (is equal and has the same format),
```